lib/concurrent/executor/abstract_executor_service.rb

```diff
@@ -10,10 +10,39 @@ module Concurrent
   class AbstractExecutorService < Synchronization::Object
     include ExecutorService
 
+    # @!visibility private
+    #Job = Struct.new(:priority, :args, :task) do
+      #def run
+        #task.call(*args)
+      #end
+      #def <=>(other)
+        #priority <=> other.priority
+      #end
+    #end
+    class Job
```

Comment:
For those interested in such things, defining this as a class rather than a struct improved performance by about 10 i/100ms in the IPS benchmark.